mimic/core.py

```diff
@@ -128,9 +153,17 @@ def entries_for_tenant(self, tenant_id, prefix_map, base_uri):
         :return: The full URI locating the service for that region
         """
         for service_id, api in self._uuid_to_api.items():
-            for entry in api.catalog_entries(tenant_id):
-                for endpoint in entry.endpoints:
-                    prefix_map[endpoint] = self.uri_for_service(
-                        endpoint.region, service_id, base_uri
-                    )
-                yield entry
+            if IExternalAPIMock.providedBy(api):
+                for entry in api.catalog_entries(tenant_id):
+                    for endpoint in entry.endpoints:
+                        prefix_map[endpoint] = api.uri_for_service(
+                            endpoint.region, service_id
+                        )
+                    yield entry
+            else:  # IAPIMock.providedBy(api)
```

Comment:
Maybe instead of having a comment here we could just add private methods, `_all_external_mocks` and `_all_service_catalog_mocks`, which explain this?  Or even an `iter_api_mocks` that yields triples of `service_id, api, external`, where `external` is just a boolean, if we don't want the overhead of having to iterate twice?